presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3Client.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.metrics.RequestMetricCollector;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.S3ClientOptions;
+import com.facebook.presto.hive.s3.HiveS3Config;
+import com.facebook.presto.hive.s3.PrestoS3FileSystemMetricCollector;
+import com.facebook.presto.hive.s3.PrestoS3FileSystemStats;
+import io.airlift.units.Duration;
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+
+public class PrestoS3Client
```

Comment:
PrestoS3Client provides AmazonS3Client for issuing the request to S3 whereas PresoS3FileSystem provides methods to operate on S3 Filesystem using the client. What are your thoughts?

What specific code repetition are you referring to? I think it will be a good idea to refactor and remove duplicated code if not required.